x-pack/plugins/apm/public/components/app/ServiceDetails/ServiceIntegrations/index.tsx

```diff
@@ -5,13 +5,15 @@
  */
 
 import { connect } from 'react-redux';
+import { getServiceDetails } from 'x-pack/plugins/apm/public/store/reactReduxRequest/serviceDetails';
 import { IReduxState } from 'x-pack/plugins/apm/public/store/rootReducer';
 import { selectIsMLAvailable } from 'x-pack/plugins/apm/public/store/selectors/license';
 import { ServiceIntegrationsView } from './view';
 
 function mapStateToProps(state = {} as IReduxState) {
   return {
-    mlAvailable: selectIsMLAvailable(state)
+    mlAvailable: selectIsMLAvailable(state),
+    serviceDetails: getServiceDetails(state)
```

Comment:
Have you thought about how we can achieve this in the future without Redux?
Eg. if we just rely on hooks inside components to do data fetching, it would be nice if we can avoid global state

Reply:
Did you look into lifting the data fetching up so we don't have to access global state?

Reply:
I haven't thought too much about this yet but yeah we'll have to find ways to solve this. I assumed it would be a combination of moving some state around and using Context for things we still want to share throughout the app without having to pass them through 2+ layers etc.